File: backend-spring-boot/src/main/java/com/neurotutor/user_service/dto/UpdateProfileRequest.java
package com.neurotutor.user_service.dto;

import com.fasterxml.jackson.annotation.JsonProperty;

public class UpdateProfileRequest {
    private String name;
    private String grade;
    private String section;
    private String avatarUrl;
    private String gender;

    public UpdateProfileRequest() {
    }

    public String getName() { return name; }
    public void setName(String name) { this.name = name; }

    public String getGrade() { return grade; }
    public void setGrade(String grade) { this.grade = grade; }

    public String getSection() { return section; }
    public void setSection(String section) { this.section = section; }

    @JsonProperty("avatar_url")
    public String getAvatarUrl() { return avatarUrl; }

    @JsonProperty("avatar_url")
    public void setAvatarUrl(String avatarUrl) { this.avatarUrl = avatarUrl; }

    public String getGender() { return gender; }
    public void setGender(String gender) { this.gender = gender; }
}

File: frontend-mobile/app/src/main/java/com/neurotutor/app/mobile/data/model/auth/UpdateProfileRequest.kt
package com.neurotutor.app.mobile.data.model.auth

import com.google.gson.annotations.SerializedName

data class UpdateProfileRequest(
    val name: String,
    val grade: String,
    val section: String,
    @SerializedName("avatar_url")
    val avatarUrl: String?,
    val gender: String
)
